src/Gui/SoFCDB.cpp

```diff
@@ -497,6 +497,43 @@ bool Gui::SoFCDB::writeToX3D(SoNode* node, bool exportViewpoints, std::string& b
     return true;
 }
 
+namespace
+{
+std::string escapeXmlAttribute(const char* str)

```

Comment:
1. Don't we have anything like that already provided? Escaping attributes should not be done manually but with some kind library - but not sure if it is available in libxerces.
2. can std::string_view be used here instead of `const char*`? It would make your life easier for the for loop